microsoft/testsuites/core/hv_module.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Logger, RemoteNode, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CentOs, Redhat
+from lisa.tools import Dmesg, Lsmod, Modinfo, Modprobe, Rpm, Uname
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers test cases previously handled by LISAv2:
+    LIS-MODULES-CHECK, VERIFY-LIS-MODULES-VERSION,
+    INITRD-MODULES-CHECK, RELOAD-MODULES-SMP
+    """,
+)
+class HvModule(TestSuite):
+    def get_modules(self, node: RemoteNode) -> list[str]:
+        """
+        Returns the hv_modules that are not directly loaded into the kernel.
+        """
+        hv_modules = [
+            "hv_storvsc",
+            "hv_netvsc",
+            "hv_vmbus",
+            "hv_utils",
+            "hid_hyperv",
+            "hv_balloon",
+            "hyperv_keyboard",
+        ]
+        uname = node.tools[Uname]
+        kernel_version = uname.get_linux_information().kernel_version_raw
+        config_path = f"/boot/config-{kernel_version}"
+
+        if node.execute(f"grep CONFIG_HYPERV_STORAGE=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_storvsc")
+        if node.execute(f"grep CONFIG_HYPERV_NET=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_netvsc")
+        if node.execute(f"grep CONFIG_HYPERV=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_vmbus")
+        if node.execute(f"grep CONFIG_HYPERV_UTILS=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_utils")
+        if node.execute(f"grep CONFIG_HID_HYPERV_MOUSE=y {config_path}").exit_code == 0:
+            hv_modules.remove("hid_hyperv")
+        if node.execute(f"grep CONFIG_HYPERV_BALLOON=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_balloon")
+        if node.execute(f"grep CONFIG_HYPERV_KEYBOARD=y {config_path}").exit_code == 0:
+            hv_modules.remove("hyperv_keyboard")
+
+        return hv_modules
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the list of given LIS kernel modules and verify if the version
+        matches with the Linux kernel release number.
+        """,
+        priority=0,
+    )
+    def verify_lis_modules_version(
+        self, case_name: str, log: Logger, node: RemoteNode
+    ) -> None:
+        if not (isinstance(node.os, CentOs) or isinstance(node.os, Redhat)):
```

Comment:
1. OS check can be added into the requirement
2. CentOs is below the RedHat. So check if it's Redhat is enough.